<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- YOLO prameters -->
    <!-- Topic Parameters -->
    <!-- cam parameters -->
    <arg name="cam2_logic" default="off"/> <!-- possible values : or, and, off-->
    <arg name="yolo_model" default="best_June.pt"/>
    <arg name="yolo_conf_thresh" default="0.6"/>
    <arg name="HFOV" default="101.662896"/>
    <arg name="VFOV" default="111.383104"/>
    <arg name="IMAGE_WIDTH" default="2047"/>
    <arg name="IMAGE_HEIGHT" default="2464"/>
    <arg name="CAMERA" default="2"/>
    <arg name="min_width" default="30"/>
    <arg name="max_width" default="400"/>
    <arg name="min_height" default="30"/>
    <arg name="max_height" default="400"/>
    <arg name="topic_name_cam" default="/ladybug/camera1/image_raw"/>
    <arg name="topic_name_cam2" default="/basler/stereo_right_anon"/>
    <arg name="topic_name_result" default="/escooter/result"/>
    <arg name="result_image_topic" default="/escooter/image"/>
    <arg name="result_image_topic2" default="/escooter/image2"/>
    <arg name="log_results" default="true"/>
    <arg name="debug" default="rviz" doc="Enable debug mode"/> <!-- This is for string [rviz, cv_image]-->

    <!-- Launch the listener node -->
    <node name="listener_node" pkg="escooter_ros" type="escooter.py" output="screen">
        <param name="yolo_model" value="$(arg yolo_model)"/>
        <param name="yolo_conf_thresh" value="$(arg yolo_conf_thresh)"/>
        <param name="HFOV" value="$(arg HFOV)"/>
        <param name="VFOV" value="$(arg VFOV)"/>
        <param name="IMAGE_WIDTH" value="$(arg IMAGE_WIDTH)"/>
        <param name="IMAGE_HEIGHT" value="$(arg IMAGE_HEIGHT)"/>
        <param name="CAMERA" value="$(arg CAMERA)"/>
        <param name="min_width" value="$(arg min_width)"/>
        <param name="max_width" value="$(arg max_width)"/>
        <param name="min_height" value="$(arg min_height)"/>
        <param name="max_height" value="$(arg max_height)"/>
        <param name="topic_name_cam" value="$(arg topic_name_cam)"/>
        <param name="topic_name_cam2" value="$(arg topic_name_cam2)"/>
        <param name="topic_name_result" value="$(arg topic_name_result)"/>
        <param name="result_image_topic" value="$(arg result_image_topic)"/>
        <param name="cam2_logic" value="$(arg cam2_logic)"/>
        <param name="log_results" value="$(arg log_results)"/>
        <param name="debug" value="$(arg debug)"/>
    </node>

    <!-- <node name="image_calibrator" pkg="ladybug5" type="image_calibrator_node" output="screen"/> -->

    <!-- <node name="ImageCalibrator" pkg="ImageCalibrator" /> -->
    <!-- <node if="$(eval arg('debug') == 'rviz')" pkg="rviz" type="rviz" name="rviz" args="-d $(find escooter_ros)/rviz/escooter.rviz"/> -->

</launch>
